### PR TITLE
Convert sponsor.js to sponsor.js, update tests to be more targeted.

### PR DESCRIPTION
**Pull request by mattgodbolt**

### CODE REVIEW

File: lib/sponsors.ts [hunk 1]
```diff
@@ -24,40 +24,39 @@
...
-    const rhsPrio = rhs.priority || 0;
+function compareSponsors(lhs: Sponsor, rhs: Sponsor): number {
+    const lhsPrio = lhs.priority;
+    const rhsPrio = rhs.priority;
     if (lhsPrio !== rhsPrio) return rhsPrio - lhsPrio;
     return lhs.name.localeCompare(rhs.name);
 }
 
-export function loadSponsorsFromString(stringConfig) {
+export function loadSponsorsFromString(stringConfig): Sponsors {
```
Comment (RubenRBS) [inline lib/sponsors.ts:53]:
```suggestion
export function loadSponsorsFromString(stringConfig: string): Sponsors {
```

File: test/sponsors-test.js [hunk 2]
```diff
@@ -22,9 +22,53 @@
 // ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 // POSSIBILITY OF SUCH DAMAGE.
 
-import {loadSponsorsFromString} from '../lib/sponsors';
+import {loadSponsorsFromString, parse} from '../lib/sponsors';
```
Comment (RubenRBS) [inline test/sponsors-test.js:25]: Weird that the pre-commit checks do not fix this before pushing

**Review by RubenRBS — APPROVED** [review]
👍

File: test/sponsors-test.js (hunk 2 above)
Comment (mattgodbolt) [reply test/sponsors-test.js:25]: which part?

Comment (mattgodbolt) [reply test/sponsors-test.js:25]: oh I see, the expanded view in GH has warnings about "at least one empty line"

**Pull request merged**